component-monitor/src/main.rs

```diff
@@ -0,0 +1,220 @@
+    consts::{SIGINT, SIGTERM},
+    iterator::Signals,
+};
+use std::{fs, path::Path, path::PathBuf, thread};
+use systemd_journal_logger::JournalLog;
+use walkdir::WalkDir;
+
+use clap::Parser;
+
+/// Manage the kernel-gpu-2404 contents
+#[derive(Debug, Parser)]
+#[command(version, about)]
+struct Cli {
+    /// The sentinel path to monitor
+    #[arg(env = "COMPONENT_SENTINEL_PATH")]
+    sentinel_path: PathBuf,
+    /// The target path to manage content in
+    #[arg(env = "COMPONENT_TARGET")]
+    target: PathBuf,
+    /// Turn on debug logging
+    #[arg(env, long, short)]
+    debug: bool,
+}
+
+/// Remove the sentinel file from target, then clear the target directory
+fn cleanup(target: &Path, sentinel_name: &str) -> Result<()> {
+    info!(target: "files", "cleaning up {target:?}");
+    let sentinel_path = target.join(sentinel_name);
+    if sentinel_path.exists() {
+        fs::remove_file(&sentinel_path)
+            .with_context(|| format!("Failed to remove {:?}", &sentinel_path))?;
+        debug!(target: "files", "removed sentinel ({:?})", &sentinel_path);
+    }
+    for entry in fs::read_dir(target).context("Failed to list target directory")? {
+        let path = entry.context("Failed to resolve entry")?.path();
+        if path.is_dir() {
+            fs::remove_dir_all(&path).with_context(|| format!("Failed to remove {:?}", &path))?;
+            debug!(target: "files", "removed {:?} recursively", &path);
+        } else {
+            fs::remove_file(&path).with_context(|| format!("Failed to remove {:?}", &path))?;
+            debug!(target: "files", "removed {:?}", &path);
+        }
+    }
+
+    Ok(())
+}
+
+/// Check if the sentinel file is current, else clean the target directory and populate again,
+/// sentinel file being last
+fn populate(source: &Path, sentinel_name: &str, target: &Path) -> Result<()> {
+    info!(target: "files", "populating {target:?} from {source:?} with sentinel {sentinel_name:?}");
+    let sentinel_src = source.join(sentinel_name);
+
+    if !sentinel_src.exists() {
+        info!(target: "files", "sentinel file ({:?}) not found, skipping", &sentinel_src);
+        return Ok(());
+    }
+
+    let sentinel_data = match fs::read(&sentinel_src) {
+        Ok(d) if d.trim_ascii() == b"" => {
+            error!(target: "files", "found empty sentinel file ({:?})", &sentinel_src);
+            return Ok(());
+        }
+        Ok(d) => d,
+        Err(e) => {
+            error!(target: "files", "failed to read sentinel file: {e}");
+            return Ok(());
+        }
+    };
+
+    let sentinel_tgt = target.join(sentinel_name);
+    if fs::read(&sentinel_tgt).is_ok_and(|content| content == sentinel_data) {
+        info!(target: "files", "found current sentinel, skipping");
+        return Ok(());
+    }
+
+    cleanup(target, sentinel_name).context("Cleanup failed")?;
+    info!(target: "files", "copying from {source:?} to {target:?}");
+    for entry in WalkDir::new(source) {
```

Comment:
This:
```suggestion
    for entry in WalkDir::new(source).min_depth(1) {
```
would allow you to remove the `if path == source` conditional below (as `source` is the only entry at depth 0)